src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -1815,3 +1867,194 @@ def encryption_protector_update(
+
+
+def managed_instance_list(
+        client,
+        resource_group_name=None):
+    '''
+    Lists managed instances in a resource group or subscription
+    '''
+
+    if resource_group_name:
+        # List all managed instances in the resource group
+        return client.list_by_resource_group(resource_group_name=resource_group_name)
+
+    # List all managed instances in the subscription
+    return client.list()
+
+
+def managed_instance_update(
+        instance,
+        administrator_login_password=None,
+        license_type=None,
+        vcores=None,
+        storage_size_in_gb=None,
+        assign_identity=False):
+    '''
+    Updates a managed instance. Custom update function to apply parameters to instance.
+    '''
+
+    # Once assigned, the identity cannot be removed
+    if instance.identity is None and assign_identity:
+        instance.identity = ResourceIdentity(type=IdentityType.system_assigned.value)
+
+    # Apply params to instance
+    instance.administrator_login_password = (
+        administrator_login_password or instance.administrator_login_password)
+    instance.license_type = (
+        license_type or instance.license_type)
+    instance.v_cores = (
+        vcores or instance.v_cores)
+    instance.storage_size_in_gb = (
+        storage_size_in_gb or instance.storage_size_in_gb)
+
+    return instance
+
+###############################################
+#                sql managed db               #
+###############################################
+
+
+def managed_db_create(
+        cmd,
+        client,
+        database_name,
+        managed_instance_name,
+        resource_group_name,
+        **kwargs):
+
+    # Determine managed instance location
+    kwargs['location'] = _get_managed_instance_location(
+        cmd.cli_ctx,
+        managed_instance_name=managed_instance_name,
+        resource_group_name=resource_group_name)
+
+    # Create
+    return client.create_or_update(
+        database_name=database_name,
+        managed_instance_name=managed_instance_name,
+        resource_group_name=resource_group_name,
+        parameters=kwargs)
+
+
+def managed_db_list(
+        client,
+        managed_instance_name,
+        resource_group_name):
+    '''
+    Lists databases in a managed instance.
+    '''
+
+    # List all databases in the server
+    return client.list_by_instance(resource_group_name=resource_group_name, managed_instance_name=managed_instance_name)
+
+
+def managed_db_restore(
+        cmd,
+        client,
+        database_name,
+        managed_instance_name,
+        resource_group_name,
+        target_managed_database_name,
+        restore_point_in_time=None,
+        **kwargs):
+    '''
+    Restores an existing managed DB (i.e. create with 'PointInTimeRestore' create mode.)
+
+    Custom function makes create mode more convenient.
+    '''
+
+    if not restore_point_in_time:
+        raise CLIError('--restore_point_in_time must be specified.')
```

Comment:
Remove `=None` so that it is required, then it will be validated at higher layer and you don't need this check

Reply:
Actually, I think you can remove `restore_point_in_time` param completely from this func so that it falls into `kwargs`. Then just make sure that it is marked as required in params.py